File: backend/internal/model/task.go
package model

import (
	"time"
)

type Task struct {
	ID          int        `gorm:"primaryKey;autoIncrement"`
	ProjectID   int        `gorm:"not null;index"`
	Title       string     `gorm:"size:200;not null"`
	Description string     `gorm:"type:text"`
	Status      string     `gorm:"size:20;default:'todo'"`
	Priority    string     `gorm:"size:20;default:'medium'"`
	AssigneeID  *int       `gorm:"index"`
	CreatedBy   int        `gorm:"not null"`
	DueDate     *time.Time `gorm:"index"`
	CompletedAt *time.Time
	IsActive    bool       `gorm:"default:true"`
	CreatedAt   time.Time  `gorm:"autoCreateTime"`
	UpdatedAt   time.Time  `gorm:"autoUpdateTime"`

	// Relationships
	Project  Project   `gorm:"foreignKey:ProjectID"`
	Assignee *User     `gorm:"foreignKey:AssigneeID"`
	Creator  User      `gorm:"foreignKey:CreatedBy"`
	Comments []Comment `gorm:"foreignKey:TaskID"`
}

// Task Status Constants
const (
	TaskStatusTodo       = "todo"
	TaskStatusInProgress = "in_progress"
	TaskStatusReview     = "review"
	TaskStatusDone       = "done"
)

// Task Priority Constants
const (
	TaskPriorityLow    = "low"
	TaskPriorityMedium = "medium"
	TaskPriorityHigh   = "high"
	TaskPriorityUrgent = "urgent"
)

type Comment struct {
	ID        int       `gorm:"primaryKey;autoIncrement"`
	TaskID    int       `gorm:"not null;index"`
	UserID    int       `gorm:"not null;index"`
	Content   string    `gorm:"type:text;not null"`
	CreatedAt time.Time `gorm:"autoCreateTime"`
	UpdatedAt time.Time `gorm:"autoUpdateTime"`

	// Relationships
	Task Task `gorm:"foreignKey:TaskID"`
	User User `gorm:"foreignKey:UserID"`
}

File: frontend/src/features/tasks/types/task.ts
import type { User } from '../../../types/api';

export type TaskStatus = 'todo' | 'in_progress' | 'review' | 'done';
export type TaskPriority = 'low' | 'medium' | 'high' | 'urgent';

export interface Task {
    id: number;
    project_id: number;
    title: string;
    description: string;
    status: TaskStatus;
    priority: TaskPriority;
    assignee_id: number | null;
    assignee?: User;
    created_by: number;
    creator?: User;
    due_date: string | null;
    completed_at: string | null;
    comment_count: number;
    is_active: boolean;
    created_at: string;
    updated_at: string;
}

export interface TaskComment {
    id: number;
    task_id: number;
    user_id: number;
    user?: User;
    content: string;
    created_at: string;
    updated_at: string;
}

export interface TaskDetail extends Task {
    comments: TaskComment[];
}

export interface CreateTaskRequest {
    project_id: number;
    title: string;
    description?: string;
    status?: TaskStatus;
    priority?: TaskPriority;
    assignee_id?: number | null;
    due_date?: string | null;
}

export interface UpdateTaskRequest {
    title?: string;
    description?: string;
    status?: TaskStatus;
    priority?: TaskPriority;
    assignee_id?: number | null;
    due_date?: string | null;
}

export interface UpdateTaskStatusRequest {
    status: TaskStatus;
}

export interface AddCommentRequest {
    content: string;
}